Signal/src/ViewControllers/AppSettings/AppSettingsViewController.m

```diff
@@ -227,45 +227,53 @@ - (void)updateTableContents
                                               }]];
 #endif
 
-    [section addItem:[OWSTableItem itemWithCustomCellBlock:^{
-        UITableViewCell *cell = [UITableViewCell new];
-        cell.preservesSuperviewLayoutMargins = YES;
-        cell.contentView.preservesSuperviewLayoutMargins = YES;
-        cell.selectionStyle = UITableViewCellSelectionStyleNone;
-
-        const CGFloat kButtonHeight = 40.f;
-        OWSFlatButton *button;
-        if (TSAccountManager.sharedInstance.isDeregistered) {
-            button = [OWSFlatButton
-                buttonWithTitle:NSLocalizedString(@"SETTINGS_REREGISTER_BUTTON", @"Label for re-registration button.")
-                           font:[OWSFlatButton fontForHeight:kButtonHeight]
-                     titleColor:[UIColor whiteColor]
-                backgroundColor:[UIColor ows_destructiveRedColor]
-                         target:self
-                       selector:@selector(reregisterUser)];
-        } else {
-            button = [OWSFlatButton buttonWithTitle:NSLocalizedString(@"SETTINGS_DELETE_ACCOUNT_BUTTON", @"")
-                                               font:[OWSFlatButton fontForHeight:kButtonHeight]
-                                         titleColor:[UIColor whiteColor]
-                                    backgroundColor:[UIColor ows_destructiveRedColor]
-                                             target:self
-                                           selector:@selector(unregisterUser)];
-        }
-        [cell.contentView addSubview:button];
-        [button autoSetDimension:ALDimensionHeight toSize:kButtonHeight];
-        [button autoVCenterInSuperview];
-        [button autoPinLeadingAndTrailingToSuperviewMargin];
-
-        return cell;
+    if (TSAccountManager.sharedInstance.isDeregistered) {
```

Comment:
Hopefully this is more readable.  In the "un-registered" state, include two buttons: "re-register" and "delete all data" (which skips the "delete account request" step of the "delete account" button).